File: src/02_algorithms/01_sorting/05_selection_sort/implementations/go/selection_sort.go
package selectionSort

func SelectionSort(array []int) {
	for i := 0; i < len(array)-1; i++ {
		min_index := i

		for j := i + 1; j < len(array); j++ {
			if array[j] < array[min_index] {
				min_index = j
			}
		}

		if min_index != i {
			array[i], array[min_index] = array[min_index], array[i]
		}
	}
}

File: src/02_algorithms/01_sorting/05_selection_sort/implementations/ts/selection_sort.ts
export const selection_sort = (array: Array<number>) => {
  for (let i = 0; i < array.length - 1; i++) {
    let min_index = i;

    for (let j = i + 1; j < array.length; j++) {
      if (array[j] < array[min_index]) {
        min_index = j;
      }
    }

    if (min_index !== i) {
      [array[i], array[min_index]] = [array[min_index], array[i]];
    }
  }
};
